docs/how-to/package-with-hatch.md

```diff
@@ -0,0 +1,380 @@
+
+## Adding code
+
+Your code should reside inside `src/<PROJECT NAME>`.
+For the calculator example we will copy `calculator.py` and `calculator.tcss` into the `src/textual_calculator` directory, so our directory will look like the following:
+
+```
+textual-calculator
+├── LICENSE.txt
+├── README.md
+├── pyproject.toml
+├── src
+│   └── textual_calculator
+│       ├── __about__.py
+│       ├── __init__.py
+│       ├── calculator.py
+│       └── calculator.tcss
+└── tests
+    └── __init__.py
+```
+
+## Adding dependencies
+
+Your Textual app will likely depend on other Python libraries (at the very least Textual itself).
+We need to list these in `pyproject.toml` to ensure that these *dependencies* are installed alongside your app.
+
+In `pyproject.toml` there should be a section beginning with `[project]`, which will look something like the following:
+
+```toml
+[project]
+name = "textual-calculator"
+dynamic = ["version"]
+description = 'A example app'
+readme = "README.md"
+requires-python = ">=3.8"
+license = "MIT"
+keywords = []
+authors = [
+  { name = "Will McGugan", email = "redacted@textualize.io" },
+]
+classifiers = [
+  "Development Status :: 4 - Beta",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: Implementation :: CPython",
+  "Programming Language :: Python :: Implementation :: PyPy",
+]
+dependencies = []
+```
+
+We are interested in the `dependencies` value, which should list the app's dependencies.
+If you want a particular version of a project you can add `==` followed by the version.
+
+For the calculator, the only dependency is Textual.
+We can add Textual by modifying the following line:
+
+```toml
+dependencies = ["textual==0.47.1"]
+```
+
+At the time of writing, the latest Textual is `0.47.1`.
+The entry in `dependencies` will ensure we get that particular version, even when newer versions are released.
+
+See the Hatch docs for more information on specifying [dependencies](https://hatch.pypa.io/latest/config/dependency/).
+
+## Environments
+
+A common problem when working with Python code, is managing multiple projects with different dependencies.
+For instance, if we had another app that used version `0.40.0` of Textual, it *may* break if we installed version `0.47.1`.
+
+The standard way of solving this is with *virtual environments* (or *venvs*), which allow each project to have its own set of dependencies.
+Hatch can create virtual environments for us, and makes working with them very easy.
+
+To create a new virtual environment, navigate to the directory with the `pyproject.toml` file and run the following command (you will only need this on first run):
```

Comment:
What does “on first run” mean?
If you know nothing about virtual environments, it's not very clear when this should be run in the process.